File: frontend/src/classe/Participe.js
export class Participe {
  constructor(etudiant, action, semestre, dateDebutParticipation, dateFinParticipation, descriptionParticipation, totalPoints = null, nbParticipation = null) {
    this._etudiant = etudiant; // Instance de Etudiant
    this._action = action;     // Instance de Action
    this._semestre = semestre; // Instance de Semestre
    this._dateDebutParticipation = dateDebutParticipation;
    this._dateFinParticipation = dateFinParticipation;
    this._descriptionParticipation = descriptionParticipation;
    this._totalPoints = totalPoints;
    this._nbParticipation = nbParticipation;
  }

  get etudiant() { return this._etudiant; }
  set etudiant(value) { this._etudiant = value; }

  get action() { return this._action; }
  set action(value) { this._action = value; }

  get semestre() { return this._semestre; }
  set semestre(value) { this._semestre = value; }

  get dateDebutParticipation() { return this._dateDebutParticipation; }
  set dateDebutParticipation(value) { this._dateDebutParticipation = value; }

  get dateFinParticipation() { return this._dateFinParticipation; }
  set dateFinParticipation(value) { this._dateFinParticipation = value; }

  get descriptionParticipation() { return this._descriptionParticipation; }
  set descriptionParticipation(value) { this._descriptionParticipation = value; }

  get totalPoints() { return this._totalPoints; }
  set totalPoints(value) { this._totalPoints = value; }

  get nbParticipation() { return this._nbParticipation; }
  set nbParticipation(value) { this._nbParticipation = value; }
}

File: backend/src/main/java/isis/projet/backend/entity/Participe.java
package isis.projet.backend.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Date;


/**
 * Entité "participe" gérant la relation ternaire entre Etudiant, Action et Semestre.
 */
@Entity
@Table(name = "participe")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Participe {

    @EmbeddedId
    private ParticipeKey id;

    /**
     * Relation vers Etudiant.
     * Ce côté est le "back" de la relation etudiant-participations.
     */
    @MapsId("idEtudiant")
    @ManyToOne
    @JoinColumn(name = "id_etudiant", nullable = false)
    @JsonBackReference(value = "etudiant-participations")
    private Etudiant etudiant;

    /**
     * Relation vers Action.
     * Ce côté est le "back" de la relation action-participations.
     */
    @MapsId("idAction")
    @ManyToOne
    @JoinColumn(name = "id_action", nullable = false)
    @JsonBackReference(value = "action-participations")
    private Action action;

    /**
     * Relation vers Semestre.
     * Ce côté est le "back" de la relation semestre-participations.
     */
    @MapsId("idSemestre")
    @ManyToOne
    @JoinColumn(name = "id_semestre", nullable = false)
    @JsonBackReference(value = "semestre-participations")
    private Semestre semestre;

    @Column(name = "totalPoints")
    private BigDecimal totalPoints;

    @Column(name = "nbParticipation")
    private Integer nbParticipation;

    @Column(name = "dateDebutParticipation")
    private Date dateDebutParticipation;

    @Column(name = "dateFinParticipation")
    private Date dateFinParticipation;

    @Column(name = "descriptionParticipation")
    @Lob
    private String descriptionParticipation;

    @Column(name = "statut")
    private Boolean  statut;

}